src/core/manager.c

```diff
@@ -3317,6 +3319,62 @@ const char *manager_get_runtime_prefix(Manager *m) {
                getenv("XDG_RUNTIME_DIR");
 }
 
+int manager_set_exec_params(Manager *m, ExecParameters *p) {
+        int r;
+
+        assert(m);
+        assert(p);
+
+        p->environment = m->environment;
+        p->confirm_spawn = manager_get_confirm_spawn(m);
+        p->cgroup_supported = m->cgroup_supported;
+
+        if (MANAGER_IS_SYSTEM(m)) {
+                r = sd_path_home(SD_PATH_SYSTEM_RUNTIME, NULL, &p->prefix[EXEC_DIRECTORY_RUNTIME]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_SYSTEM_STATE_PRIVATE, NULL, &p->prefix[EXEC_DIRECTORY_STATE]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_SYSTEM_STATE_CACHE, NULL, &p->prefix[EXEC_DIRECTORY_CACHE]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_SYSTEM_STATE_LOGS, NULL, &p->prefix[EXEC_DIRECTORY_LOGS]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_SYSTEM_CONFIGURATION, NULL, &p->prefix[EXEC_DIRECTORY_CONF]);
+                if (r < 0)
+                        return r;
+
+        } else {
+                r = sd_path_home(SD_PATH_USER_RUNTIME, NULL, &p->prefix[EXEC_DIRECTORY_RUNTIME]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_USER_CONFIGURATION, NULL, &p->prefix[EXEC_DIRECTORY_STATE]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_USER_STATE_CACHE, NULL, &p->prefix[EXEC_DIRECTORY_CACHE]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_USER_CONFIGURATION, "log", &p->prefix[EXEC_DIRECTORY_LOGS]);
+                if (r < 0)
+                        return r;
+
+                r = sd_path_home(SD_PATH_USER_CONFIGURATION, NULL, &p->prefix[EXEC_DIRECTORY_CONF]);
+                if (r < 0)
+                        return r;
```

Comment:
I think we should have two tables for this, and then iterate trhough them, i.e. something like this:

```c
static const struct {
        ExecDirectory directory;
        uint64_t type;
        const char *suffix;
} paths[_EXEC_DIRECTORY_MAX] = {
        { EXEC_DIRECTORY_RUNTIME, SD_PATH_RUNTIME, NULL },
        …
};
```

We'd just need one of these tables for the user and one for the system case, and could then use a single loop for applying them, and storing them in a "prefix" array in the Manager object.